lib/include/openturns/OTishigamiusecase.hxx

```diff
@@ -0,0 +1,176 @@
+      s12_ = 0.0;  // Interaction of (X1, X2)
+      // Interaction of (X1, X3)
+      s13_ = pow(b_, 2.0) * pow(M_PI, 8.0) * (1.0 / 9.0 - 1.0 / 25.0) / 2.0 / variance_;
+      s23_ = 0.0;  // Interaction of (X2, X3)
+      s123_ = 0.0;  // Interaction of (X1, X2, X3)
+      sT1_ = s1_ + s13_;  // Total of X1
+      sT2_ = s2_;  // Total of X2
+      sT3_ = s3_ + s13_;  // Total of X3
+      
+    }
+    Function getModel()
+    {
+      return model_;
+    }
+    UnsignedInteger getDimension()
+    {
+      return dimension_;
+    }
+    Scalar getA()
+    {
+      return a_;
+    }
+    Scalar getB()
+    {
+      return b_;
+    }
+    JointDistribution getInputDistribution()
+    {
+      return inputDistribution_;
+    }
+    Scalar getMean()
+    {
+      return mean_;
+    }
+    Scalar getVariance()
+    {
+      return variance_;
+    }
+    Point getFirstOrderSobolIndices()
+    {
+      const Point firstOrderIndices({s1_, s2_, s3_});
+      return firstOrderIndices;
+    }
+    Point getTotalSobolIndices()
+    {
+      const Point totalIndices({sT1_, sT2_, sT3_});
+      return totalIndices;
+    }
+    Scalar getFirstOrderInteractionSobolIndex(const Indices indices)
+    {
+      Scalar value = -1.0;
+      if (indices.getSize() == 2)
+      {
+        if (indices[0] == 0 && indices[1] == 1) value = s12_;
+        else if (indices[0] == 0 && indices[1] == 2) value = s13_;
+        else if (indices[0] == 1 && indices[1] == 2) value = s23_;
+        else
+          throw InvalidArgumentException(HERE) << "Inconsistent 2D indices = " << indices;
+      }
+      else if (indices.getSize() == 3)
+        if (indices[0] == 0 && indices[1] == 1 && indices[2] == 2) value = s123_;
+        else
+          throw InvalidArgumentException(HERE) << "Inconsistent 3D indices = " << indices;
+      else
+        throw InvalidArgumentException(HERE) << "Inconsistent dimension of indices = " << indices.getSize();
+      return value;
+    }
+    Scalar getTotalInteractionSobolIndex(const Indices indices)
+    {
+      Scalar value = -1.0;
+      if (indices.getSize() == 2)
+      {
+        if (indices[0] == 0 && indices[1] == 1) value = s12_ + s123_;
+        else if (indices[0] == 0 && indices[1] == 2) value = s13_+ s123_;
+        else if (indices[0] == 1 && indices[1] == 2) value = s23_+ s123_;
+        else
+          throw InvalidArgumentException(HERE) << "Inconsistent 2D indices = " << indices;
+      }
+      else if (indices.getSize() == 3)
+        if (indices[0] == 0 && indices[1] == 1 && indices[2] == 2) value = s123_;
+        else
+          throw InvalidArgumentException(HERE) << "Inconsistent 3D indices = " << indices;
+      else
+        throw InvalidArgumentException(HERE) << "Inconsistent dimension of indices = " << indices.getSize();
+      return value;
+    }
+
+  private:
+    Function model_;
+    JointDistribution inputDistribution_;
+    UnsignedInteger dimension_;
+    Scalar a_;
+    Scalar b_;
```

Comment:
default values of dim, a, b can be initialized here

Reply:
Ok. Fixed at 4578d84604266a4e6d4eb217f0aec3dfd83bc6bb